src/ruby/ext/symengine/ruby_basic.c

```diff
@@ -0,0 +1,109 @@
+#include "ruby_basic.h"
+
+static void cbasic_free(void *ptr){
+    basic_struct *basic_ptr = ptr;
+    basic_free(basic_ptr);
+}
+
+static VALUE cbasic_alloc(VALUE klass){
```

Comment:
I had referred to [this](http://guides.rubygems.org/gems-with-extensions/#c-extension) guide for declaring functions. I will give a try without `static` and post the results.